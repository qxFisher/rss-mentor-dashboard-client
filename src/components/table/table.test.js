import React from 'react';
import Table from './table';
import renderer from 'react-test-renderer';
import {render, cleanup} from 'react-testing-library';

afterEach(cleanup);

const testData = {
    "tasks": [
      {
        "taskName": "Code Jam \"CV\"",
        "taskLink": "https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-cv.md",
        "taskStatus": "Checked"
      },
      {
        "taskName": "Code Jam \"CoreJS\"",
        "taskLink": "https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-corejs.md",
        "taskStatus": "Checked"
      },],
    "mentors": [
      {
        "mentorName": "Aliaksandr Zayats",
        "mentorGithub": "alex-zayats",
        "students": [
          {
            "studentGithub": "uniorunr",
            "checkedTasks": [
              "Markup #1",
              "Code Jam \"DOM, DOM Events\""
            ]
          },
          {
            "studentGithub": "alexeydavydov",
            "checkedTasks": []
          },
          {
            "studentGithub": "melllivoracapensis",
            "checkedTasks": [
              "Markup #1",
              "Code Jam \"DOM, DOM Events\""
            ]
          },
          {
            "studentGithub": "soulshaker",
            "checkedTasks": [
              "Markup #1",
              "Code Jam \"DOM, DOM Events\""
            ]
          }
        ]
      },
    ],
};

const testSelectedOption = {value: "alex-zayats", label: "Aliaksandr Zayats (alex-zayats)"};

it('Check the existing of table', () => {
  render(
    <Table data={testData} selectedOption={testSelectedOption}/>,
  );
  
  const table = document.querySelector('table');
  expect(table).toBeTruthy();
});

it('Check the count of rows with the given data', () => {
  render(
    <Table data={testData} selectedOption={testSelectedOption}/>,
  );
  
  const table = document.querySelector('table');
  const tableRows = table.querySelectorAll('tr');

  expect(tableRows.length).toBe(5);
});

it('Check the count of cells', () => {
  render(
    <Table data={testData} selectedOption={testSelectedOption}/>,
  );
  
  const table = document.querySelector('table');
  const tableCells = table.querySelectorAll('td').length;
  expect(tableCells).toBe(12);
});

test('Table to match snapshot', () => {
  const component = renderer.create(
    <Table data={testData} selectedOption={testSelectedOption}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
