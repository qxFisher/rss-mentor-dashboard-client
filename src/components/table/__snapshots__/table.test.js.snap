// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table changes the cells when new option selected 1`] = `
<table
  className="dashboard-table"
>
  <thead>
    <tr>
      <th />
      <th>
        <a
          href="https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-cv.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          Code Jam "CV"
        </a>
      </th>
      <th>
        <a
          href="https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-corejs.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          Code Jam "CoreJS"
        </a>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a
          href="https://github.com/uniorunr"
          rel="noopener noreferrer"
          target="_blank"
        >
          uniorunr
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/alexeydavydov"
          rel="noopener noreferrer"
          target="_blank"
        >
          alexeydavydov
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/melllivoracapensis"
          rel="noopener noreferrer"
          target="_blank"
        >
          melllivoracapensis
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/soulshaker"
          rel="noopener noreferrer"
          target="_blank"
        >
          soulshaker
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
  </tbody>
</table>
`;

exports[`Table to match snapshot 1`] = `
<table
  className="dashboard-table"
>
  <thead>
    <tr>
      <th />
      <th>
        <a
          href="https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-cv.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          Code Jam "CV"
        </a>
      </th>
      <th>
        <a
          href="https://github.com/rolling-scopes-school/tasks/blob/2018-Q3/tasks/codejam-corejs.md"
          rel="noopener noreferrer"
          target="_blank"
        >
          Code Jam "CoreJS"
        </a>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a
          href="https://github.com/uniorunr"
          rel="noopener noreferrer"
          target="_blank"
        >
          uniorunr
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/alexeydavydov"
          rel="noopener noreferrer"
          target="_blank"
        >
          alexeydavydov
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/melllivoracapensis"
          rel="noopener noreferrer"
          target="_blank"
        >
          melllivoracapensis
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
    <tr>
      <td>
        <a
          href="https://github.com/soulshaker"
          rel="noopener noreferrer"
          target="_blank"
        >
          soulshaker
        </a>
      </td>
      <td
        className="Checked"
      />
      <td
        className="Checked"
      />
    </tr>
  </tbody>
</table>
`;
